src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -664,6 +681,37 @@ private static bool ShowInStackTrace(MethodBase mb)
             return !(mb.IsDefined(typeof(StackTraceHiddenAttribute)) || (mb.DeclaringType?.IsDefined(typeof(StackTraceHiddenAttribute)) ?? false));
         }
 
+        private static bool TryResolveStateMachineMethod(ref MethodBase mb, ref Type declaringType)
+        {
+            Debug.Assert(mb != null);
+            Debug.Assert(declaringType != null);
+
+            Type dt = declaringType.DeclaringType;
+            if (dt == null)
+            {
+                return false;
+            }
+
+            MethodInfo[] methods = dt.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+            if (methods == null)
+            {
+                return false;
+            }
+
+            foreach (MethodInfo method in methods)
+            {
+                StateMachineAttribute asma = method.GetCustomAttribute<StateMachineAttribute>();
+                if (asma != null && asma.StateMachineType == declaringType && method is MethodBase matched)
```

Comment:
>method is MethodBase matched [](start = 78, length = 28)

MethodInfo derives from MethodBase, so this is always true, isn't it?

Reply:
Previously I used `var` so didn't notice this; the dangers of var or the benefits of declaring types...

Reply:
removed